app/src/org/commcare/android/adapters/WiFiDirectUIController.java

```diff
@@ -0,0 +1,114 @@
+package org.commcare.android.adapters;
+
+import android.view.View;
+
+import org.commcare.activities.CommCareWiFiDirectActivity;
+import org.commcare.dalvik.R;
+import org.commcare.interfaces.CommCareActivityUIController;
+import org.commcare.views.SquareButtonWithText;
+import org.javarosa.core.services.locale.Localization;
+
+public class WiFiDirectUIController implements CommCareActivityUIController {
+    private SquareButtonWithText sendButton;
+    private SquareButtonWithText submitButton;
+    private SquareButtonWithText discoverButton;
+    private SquareButtonWithText modeButton;
+
+    private final CommCareWiFiDirectActivity activity;
+
+    public WiFiDirectUIController(CommCareWiFiDirectActivity activity) {
+        this.activity = activity;
+    }
+
+    @Override
+    public void setupUI() {
+        setupButtonListeners();
+    }
+
+    private void setupButtonListeners() {
+        modeButton = (SquareButtonWithText)activity.findViewById(R.id.mode);
```

Comment:
Now that we have a pattern for our controllers we should see what we can do to use the annotation style harnessing that is in the old Activitiy framework to get rid of some of this verbosity.